### PR TITLE
kube-vip-cloud-provider: add package

### PR DESCRIPTION
**Pull request by guptavinayak11**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

New spec file created for kube-vip-cloud-provider. This package is used for AksIoT and a container image with the package must be built by Microsoft.

###### Change Log  

- New spec file for kube-vip-cloud-provider

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Built locally with the Mariner Toolchain.

BuddyBuild:
- AMD64 - [236835](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=236835&view=logs&j=75cba72d-09cb-503a-3927-4ab61ac4a179&t=246631ef-4e63-576d-3b21-bc52cd4b0a93&l=2133)
- ARM64 - [236867](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=236867&view=results)

### CODE REVIEW

**Comment by mfrw** [comment]
I've changed the commit message to be in the `imperative` mood.
I would encourage giving this a read whenever convenient: [git-commit](https://cbea.ms/git-commit/)

**Review by lilustga — APPROVED** [review]
looks good to me

File: SPECS/kube-vip-cloud-provider/kube-vip-cloud-provider.signatures.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "Signatures": {
+     "kube-vip-cloud-provider-0.0.2-vendor.tar.gz": "6dda8a2eb13a1c29f5f5c70deaa7f399e6961c221d9c8ee3b0323c9d99ee5a12",
+     "kube-vip-cloud-provider-0.0.2.tar.gz": "f8cfdff062f55c5e167503e1fc1ca0e99f9d53858660a3be07858a3db680a7e5"
```
Comment (mfrw) [inline SPECS/kube-vip-cloud-provider/kube-vip-cloud-provider.signatures.json:4]: These files do not seem to present in the blob store :)

Comment (guptavinayak11) [reply SPECS/kube-vip-cloud-provider/kube-vip-cloud-provider.signatures.json:4]: Yes, I will add them in the blob store, by today.

File: SPECS/kube-vip-cloud-provider/kube-vip-cloud-provider.spec [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+Release:        1%{?dist}
+License:        ASL 2.0
+URL:            https://github.com/kube-vip/kube-vip-cloud-provider
+Group:          Applications/Text
+Vendor:         Microsoft
+Distribution:   Mariner
+Source0:        https://github.com/kube-vip/%{name}/archive/refs/tags/v%{version}.tar.gz#/%{name}-%{version}.tar.gz
+#Note that the source file should be renamed to the format {name}-%{version}.tar.gz
+
+# Steps to manually created vendor tarball, no download link.
```
[the remark this answers is not in the document]
Comment (guptavinayak11) [reply SPECS/kube-vip-cloud-provider/kube-vip-cloud-provider.spec]: Done

File: SPECS/kube-vip-cloud-provider/kube-vip-cloud-provider.signatures.json (hunk 1 above)
Comment (guptavinayak11) [reply SPECS/kube-vip-cloud-provider/kube-vip-cloud-provider.signatures.json:4]: Can you try now? :-)

Comment (mfrw) [reply SPECS/kube-vip-cloud-provider/kube-vip-cloud-provider.signatures.json:4]: We recently did a change that allows anyone to run a `BuddyBuild`.
Can you please include the `BuddyBuild` Pipeline id for both AMD64 & ARM64 for your changes.
Thanks.

> Can you try now? :-)

Sure, this is a representative example for AMD64 BuddyBuild of this PR: [PR-3728](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=236835&view=logs&j=75cba72d-09cb-503a-3927-4ab61ac4a179&t=246631ef-4e63-576d-3b21-bc52cd4b0a93&l=2133)

Please run it similarly for ARM64 [for your other change as well :)]
Also, I added this specific detail in the description of the PR under the testing methodology.

Comment (guptavinayak11) [reply SPECS/kube-vip-cloud-provider/kube-vip-cloud-provider.signatures.json:4]: Added this. Thanks for the information.

File: SPECS/kube-vip-cloud-provider/kube-vip-cloud-provider.spec [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+%install
+install -d %{buildroot}%{_bindir}
+install kube-vip-cloud-provider %{buildroot}%{_bindir}/kube-vip-cloud-provider
+
+%files
+%{_bindir}/kube-vip-cloud-provider
+
+%changelog
+* Mon Sep 06 2022 Vinayak Gupta <guptavinayak@microsoft.com> 0.0.2-1
```
Comment (mfrw) [inline SPECS/kube-vip-cloud-provider/kube-vip-cloud-provider.spec:45]:
```suggestion
* Tue Sep 06 2022 Vinayak Gupta <guptavinayak@microsoft.com> - 0.0.2-1
```

**Review by mfrw — COMMENTED** [review]
nit in `%changelog`

File: SPECS/kube-vip-cloud-provider/kube-vip-cloud-provider.spec (hunk 3 above)
Comment (guptavinayak11) [reply SPECS/kube-vip-cloud-provider/kube-vip-cloud-provider.spec]: Done

File: SPECS/kube-vip-cloud-provider/kube-vip-cloud-provider.spec [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+%install
+install -d %{buildroot}%{_bindir}
+install kube-vip-cloud-provider %{buildroot}%{_bindir}/kube-vip-cloud-provider
+
+%files
+%{_bindir}/kube-vip-cloud-provider
+
+%changelog
+* Tue Sep 06 2022 Vinayak Gupta <guptavinayak@microsoft.com> 0.0.2-1
```
Comment (mfrw) [inline SPECS/kube-vip-cloud-provider/kube-vip-cloud-provider.spec:45]:
```suggestion
* Tue Sep 06 2022 Vinayak Gupta <guptavinayak@microsoft.com> - 0.0.2-1
```

Comment (guptavinayak11) [reply SPECS/kube-vip-cloud-provider/kube-vip-cloud-provider.spec]: Done

**Review by mfrw — APPROVED** [review]
LGTM 🥇

**Pull request merged**